pkg/model/iam/iam_builder.go

```diff
@@ -236,17 +259,24 @@ func (b *PolicyBuilder) BuildAWSPolicy() (*Policy, error) {
 	return p, nil
 }
 
+func NewPolicy(clusterName string) *Policy {
```

Comment:
Would it make sense for this to be a receiver on `*PolicyBuilder`? It could then store the `PolicyBuilder` in the `Policy` and a lot of receivers on `*PolicyBuilder` could instead be on `*Policy`.

I think `PolicyBuilder` might be better named `PolicyFactory`.